data/cookbook/debug-print-a-value/00-ppx_deriving.ml

```diff
@@ -0,0 +1,68 @@
+---
+packages:
+- name: "ppx_deriving"
+  tested_version: "6.0.3"
+  used_libraries:
+  - ppx_deriving
+discussion: |
+    It is required to have `(preprocess (pps ppx_deriving.show))` in `dune` file.
+---
+
+(* string *)
+let () = print_endline "hello world"
+let () = print_endline ([%show: string] "hello world")
+
+(* integer *)
+let show_int = [%show: int]
+let () = print_endline (show_int 7)
+
+(* tuple of integer and string *)
+let show_pair : (int * string) -> string = [%show: (int * string)]
+let () = print_endline (show_pair (3, "hello"))
+
+(* list of tuples, each is a boolean and character *)
+let () = print_endline @@ [%show: (bool * char) list] [ (true, 'a'); (false, 'b') ]
+
+(* user-defined type; binary tree with weighted vertices *)
+type tree =
+    | Leaf of float
+    | Node of float * tree * tree
+    [@@deriving show]
```

Comment:
We might want to explain why we are suddenly switching from `[%show: ...]` to `[@@deriving show]`.

Reply:
It helps to take inspiration from the JSON parsing recipes, to give a better explanation of what `@@deriving` does.

And indeed, makes sense to add a single sentence that explains that `[%show: t]` generates the code to print values of type `t`

Reply:
@sabine would you recommend me background resources to check? maybe I can write a short explanation and enhance the tutorial's accessibility.